src/lib/features/snackbar/model/store.ts

```diff
@@ -0,0 +1,47 @@
+import { writable, get } from 'svelte/store';
+import type { SnackbarOptions } from './types';
+
+/**
+ * The queue of different snackbars. It works like this:
+ *
+ * • First comes served – the first snackbar stays on the screen for 4 seconds
+ * • Next snackbar waits in the queue for its turn
+ * • Snackbar timing can be prolonged – if the time of the first snackbar in the queue
+ * didn't pass but the call for this type of snackbar fired again, timeout resets
+ * to 4 seconds
+ * • Pressing on the Snackbar's action removes it from the queue
+ */
+export const snackbarQueue = writable<SnackbarOptions[]>([]);
+
+const SNACKBAR_TIMEOUT = 4000;
+let timeout: ReturnType<typeof setTimeout>;
+
+// TODO: экспортировать фильтр и сделать обертку для каждого колбека
+
+export const filterQueueByText = (text: string) => {
+  snackbarQueue.update((queue) => {
+    return queue.filter((snack) => snack.text !== text);
```

Comment:
issue (non-blocking): this is not as efficient as it could be, since this function will only ever be used to remove the first snackbar in the queue (from what it seems). But since this queue is unlikely to contain more than 3 elements at a time, it's not a big deal.

Reply:
Also, I don't understand when people write `(x) => { return ...; }` instead of simply `(x) => ...`, which is what arrow functions are mainly for

Reply:
@aabounegm that sounds kind of harsh